reports/tests.py

```diff
@@ -0,0 +1,521 @@
+        from other instances are not included in the subtotals, total, etc.
+        """
+        desired_instance = OpenEdXInstanceFactory()
+        make_test_appserver(desired_instance, status=AppServerStatus.Running)
+        make_test_appserver(desired_instance, status=AppServerStatus.Terminated)
+        make_test_appserver(desired_instance, status=AppServerStatus.ConfigurationFailed)
+
+        another_instance = OpenEdXInstanceFactory()
+        make_test_appserver(instance=another_instance)
+
+        invoice_month = self._generate_invoice_date()
+        expected_charges_details = {
+            'name': 'Mock AppServer charge',
+            'billing_start': invoice_month,
+            'billing_end': invoice_month,
+            'days': 20,
+            'charge': 12,
+        }
+        charges_details_mock.return_value = expected_charges_details
+        appservers_charges, appservers_total = get_instance_charges(desired_instance, invoice_month)
+
+        self.assertEqual(len(appservers_charges), 2)
+        for charge in appservers_charges:
+            self.assertEqual(charge, expected_charges_details)
+
+        self.assertEqual(appservers_total, 2 * expected_charges_details['charge'])
+
+    def test_generate_watched_forks_instances(self):
+        """
+        This will test that the method is able to return all of the watched
+        forks' instances for a specific organization and not anything else.
+        """
+        fork1, fork2, fork3 = self._setup_watched_forks()
+        watched_forks = generate_watched_forks_instances(self.organization)
+
+        # Because the third fork doesn't have any instances, it's not going to be shown in the reports
+        self.assertIsInstance(watched_forks, defaultdict)
+        self.assertEqual(len(watched_forks), 2)
+
+        # Watched Fork 1 has only two instances
+        self.assertIsInstance(watched_forks[fork1], list)
+        self.assertEqual(len(watched_forks[fork1]), 2)
+        for instance in watched_forks[fork1]:
+            self.assertIsInstance(instance, OpenEdXInstance)
+
+        # Watched Fork 2 has only one instance
+        self.assertIsInstance(watched_forks[fork2], list)
+        self.assertEqual(len(watched_forks[fork2]), 1)
+        for instance in watched_forks[fork2]:
+            self.assertIsInstance(instance, OpenEdXInstance)
+
+        # Watched Fork 3 has no instances
+        self.assertIsInstance(watched_forks[fork3], list)
+        self.assertEqual(len(watched_forks[fork3]), 0)
+
+    @mock.patch('reports.helpers.get_instance_charges')
+    def test_generate_charges(self, instance_charges_mock):
+        """
+        Make sure that we're able to generate charges for all of the appservers
+        that belongs to any fork under a given organization.
+        The test here will pass if all forks with their AppServers are included
+        in the billing data each associated with all charges that belong to its instances.
```

Comment:
Should there be a comma before `each`? If you mean that „each [should be] associated with all charges that belong to the instances“, you can also add an `and` („the test will pass if […] and each one is associated with all charges […]“).